lib/active_merchant/billing/gateways/forte.rb

```diff
@@ -153,24 +149,24 @@ def add_customer_paymethod(post, payment_method)
       end
 
       def add_customer_billing_address(post, options)
+        return unless address == options[:billing_address] || options[:address]
+
         post[:addresses] = []
-        if address = options[:billing_address] || options[:address]
-          billing_address = {}
-          billing_address[:address_type] = "default_billing"
-          billing_address[:physical_address] = {}
-          billing_address[:physical_address][:street_line1] = address[:address1] if address[:address1]
-          billing_address[:physical_address][:street_line2] = address[:address2] if address[:address2]
-          billing_address[:physical_address][:postal_code] = address[:zip] if address[:zip]
-          billing_address[:physical_address][:region] = address[:state] if address[:state]
-          billing_address[:physical_address][:locality] = address[:city] if address[:city]
-          billing_address[:email] = options[:email] if options[:email]
-          post[:addresses] << billing_address
-        end
+        billing_address = {}
+        billing_address[:address_type] = "default_billing"
+        billing_address[:physical_address] = {}
+        billing_address[:physical_address][:street_line1] = address[:address1] if address[:address1]
+        billing_address[:physical_address][:street_line2] = address[:address2] if address[:address2]
+        billing_address[:physical_address][:postal_code] = address[:zip] if address[:zip]
+        billing_address[:physical_address][:region] = address[:state] if address[:state]
+        billing_address[:physical_address][:locality] = address[:city] if address[:city]
+        billing_address[:email] = options[:email] if options[:email]
+        post[:addresses] << billing_address
       end
 
       def add_billing_address(post, payment, options)
         post[:billing_address] = {}
-        if address = options[:billing_address] || options[:address]
+        if address == options[:billing_address] || options[:address]
```

Comment:
I think this was meant to be assignment.  Does it need parentheses to work correctly?

Reply:
Ugh, my bad. You're right. I literally forgot you could do this in Ruby.